tiledb/sm/subarray/subarray.cc

```diff
@@ -1121,16 +1121,28 @@ Status Subarray::set_coalesce_ranges(bool coalesce_ranges) {
 }
 
 Status Subarray::to_byte_vec(std::vector<uint8_t>* byte_vec) const {
```

Comment:
This gets used as part of `serialize_subarray` and `deserialize_subarray` when serializing a dense query. It looks like:
1. The labels are getting dropped from the `serialize_subarray`/`deserialize_subarray` round trip.
2. It doesn't matter, because we replace the subarray with a fully deserialized subarray using the `subarray_to_capnp` and `subarray_from_capnp` functions. (Right now this is guaranteed due to the below changes in `num_range`.) 

It looks like you should be able to fully remove the `serialize_subarray`, `deseriaze_subarray`, and `Subarray::to_byte_vec` functions. For the `writer_to_capnp`, I recommend we always serialize and deserialize the subarray just like we do in the reader path using `subarray_to_capnp` and `subarray_from_capnp` since it is safer than having special functions for edge cases.